qa/full-cluster-restart/src/test/java/org/opensearch/upgrades/PluginInfoIT.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.upgrades;
+
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.test.rest.yaml.ObjectPath;
+
+import java.util.Map;
+
+public class PluginInfoIT extends AbstractFullClusterRestartTestCase {
+    public void testPluginInfoSerialization() throws Exception {
+        // Ensure all nodes are able to come up, validate with GET _nodes.
+        Response response = client().performRequest(new Request("GET", "_nodes"));
+        ObjectPath objectPath = ObjectPath.createFromResponse(response);
+        final Map<String, Object> nodeMap = objectPath.evaluate("nodes");
+        // Any issue in PluginInfo serialization logic will result into connection failures
+        // and hence reduced number of nodes.
+        assertEquals(2, nodeMap.keySet().size());
```

Comment:
Nice, can we build on this assertion to check which plugins we expected to come up are available?

For the nodes that did not come online, can we do any more detailed validation to be sure why they didn't come up, I'm not clear why this test expects two nodes not to be available.

Reply:
@saratvemulapalli Would you mind doing a pass over these tests to see if they captured what you were looking for?

Reply:
This is good, at least it says the wire compatibility works. 
Ideally the test we need is de-serialize pluginInfo on new nodes to verify we can really read them. But since we need to add some more PRs, I'd be ok to see that test in a follow up PR, since this change has been there for a while.

Also why do we see 2 nodes with restart upgrade? Shouldn't all 4 nodes come up successfully?

Reply:
full-cluster-restart tests use a 2 node cluster - https://github.com/opensearch-project/OpenSearch/blob/main/qa/full-cluster-restart/build.gradle#L46.

mixed-cluster tests use a 4 node cluster.

Reply:
I'd love to read this in from a class/configuration, but this looks like a great foundation - resolving this thread.